sonar-slang-plugin/src/test/java/com/sonarsource/slang/kotlin/KotlinParserTest.java

```diff
@@ -90,12 +92,20 @@ public void isExpressions() {
   @Test
   public void testFunctionDeclaration() {
     FunctionDeclarationTree functionDeclarationTree = ((FunctionDeclarationTree) kotlin("fun function1(a: Int, b: String): Boolean { true; }"));
-    // FIXME test return type, parameter names, modifiers
     assertTree(functionDeclarationTree.name()).isIdentifier("function1").hasTextRange(1, 4, 1, 13);
+    assertTree(functionDeclarationTree.returnType()).isIdentifier("Boolean");
     assertThat(functionDeclarationTree.formalParameters()).hasSize(2);
+    assertTree(functionDeclarationTree.formalParameters().get(1)).isIdentifier("b");
     assertTree(functionDeclarationTree.body()).isBlock(LiteralTree.class);
-
-    assertTree(((FunctionDeclarationTree) kotlin("fun function1(a: Int, b: String): Boolean = true")).body()).isNotNull();
+    assertThat(functionDeclarationTree.modifiers()).isEmpty();
+
+    FunctionDeclarationTree functionWithModifier = (FunctionDeclarationTree) kotlin("internal fun function1(a: Int, b: String): Boolean = true");
+    assertTree(functionWithModifier.body()).isNotNull();
+    NativeKind expectedModifierKind = new KotlinNativeKind(LeafPsiElement.class, "internal");
+    assertThat(functionWithModifier.modifiers()).hasSize(1);
+    Tree modifierTree = functionWithModifier.modifiers().get(0);
+    assertTree(modifierTree).isInstanceOf(NativeTree.class);
+    assertThat(((NativeTree) modifierTree).nativeKind()).isEqualTo(expectedModifierKind);
```

Comment:
Same thing as in the other PR: I would prefer to have a test on syntactic equivalence so that I can understand why we need a different native kind.

Reply:
Done (bd329faf3c92cc56cc352e96e667a1ce9a5c06fa)